src/javascript/app_2/Modules/Trading/Components/Form/TradeParams/Duration/duration.jsx

```diff
@@ -165,26 +166,31 @@ const Duration = ({
                             market_close_times={market_close_times}
                             number_input_props={props.number_input}
                             onChange={onChange}
+                            onChangeUiStore={onChangeUiStore}
                             server_time={server_time}
                             sessions={sessions}
                             shared_input_props={props.shared_input}
                             start_date={start_date}
                             start_time={start_time}
-                            onChangeUiStore={onChangeUiStore}
-                            duration_t={duration_t}
                             symbol={symbol}
                         /> }
                     { !is_advanced_duration &&
                         <SimpleDuration
                             getDurationFromUnit={getDurationFromUnit}
                             changeDurationUnit={changeDurationUnit}
+                            duration_min_max={duration_min_max}
+                            duration_t={duration_t}
                             duration_units_list={duration_units_list}
+                            expiry_date={expiry_date}
+                            is_nativepicker={is_nativepicker}
                             number_input_props={props.number_input}
                             onChange={onChange}
+                            onChangeUiStore={onChangeUiStore}
+                            server_time={server_time}
                             shared_input_props={props.shared_input}
                             simple_duration_unit={simple_duration_unit}
-                            onChangeUiStore={onChangeUiStore}
-                            duration_t={duration_t}
+                            start_date={start_date}
+                            start_time={start_time}
```

Comment:
To me it seems as if we are passing the point of passing down too many props through `<Duration/>` to `<SimpleDuration />` and `<AdvancedDuration />`, to prevent all some of this props drilling we could wrap `<DatePicker>` in a new component that is directly connected to the tradeStore? What do you think? @kellybinary @zakibinary @negar-binary 

https://blog.kentcdodds.com/prop-drilling-bb62e02cb691

Reply:
imo if we put `observer(datepicker)` as props in `datepicker` component  we can reduce the props being passed from `duration`.

Reply:
agreed, better to add break it down into another component. in there, we can pass common props from trade_store such as `min_date`, `max_date`, `disable_trading_events`, `is_nativepicker` etc..

Reply:
there are common `7 props` from trade store being passed from `advance duration` and `simple duration` to datepicker. which is quite a lot

Reply:
yea.. more reason to break it down :P

Reply:
DatePicker already has an `observer()`. I was thinking to add a new component for trading datepicker in `Trading/Components/` (might need to do this for timepicker as well - not sure of the use case yet)

Reply:
thank you for the review 🙏 , i will create new component where it extends the datepicker, and calculate the necessary min date and max date from there instead of having the same for advance and simple. therefore, the datepicker wrapper will pass observer from the tradestore.

Reply:
Since you have the `<DatepickerWrapper>` connected directly to the tradeStore, you can remove the props that aren't used from here in `<SimpleDuration>` and above in `<AdvancedDuration>`